six/modules/c++/six/include/six/ReadControl.h

```diff
@@ -108,12 +109,43 @@ class ReadControl
      *  This function reads in the image area specified by the region.
      *  If you want us to use a work buffer, you can set it through region.
      *
+     *  To simply read the entire image, pass a default-constructed
+     *  Region as the first parameter
+     *
      *  Once read, the image buffer is set in both the region pointer,
      *  and in the return value, for convenience
+     *
+     *  For safety, prefer the overload below.
      */
     virtual UByte* interleaved(Region& region, size_t imageNumber) = 0;
 
     /*!
+     *  This function reads in the image area specified by the region.
+     *  If you want us to use a work buffer, you can set it through region.
```

Comment:
Since `buffer` takes ownership of this, spell out in the comments here what happens if you do in fact set that work buffer through region (i.e. you don't own the memory after this call - the ScopedArray does)